debezium-core/src/main/java/io/debezium/pipeline/ChangeEventSourceCoordinator.java

```diff
@@ -193,11 +196,37 @@ protected void executeChangeEventSources(CdcSourceTaskContext taskContext, Snaps
         LOGGER.debug("Snapshot result {}", snapshotResult);
 
         if (running && snapshotResult.isCompletedOrSkipped()) {
+            if (snapshotResult.isCompleted()) {
+                delayStreamingIfNeeded(context);
+            }
             previousLogContext.set(taskContext.configureLoggingContext("streaming", partition));
             streamEvents(context, partition, snapshotResult.getOffset());
         }
     }
 
+    /**
+     * Delays streaming execution as per the {@link CommonConnectorConfig#STREAMING_DELAY_MS} parameter.
+     */
+    protected void delayStreamingIfNeeded(ChangeEventSourceContext context) throws InterruptedException {
+        Duration streamingDelay = connectorConfig.getStreamingDelay();
```

Comment:
I believe it is necessary to check if streaming is enabled on `Snapshotter` and skip delay if it is not.

Reply:
Thanks for the review @jpechane 
Addressed the comments.

Reply:
I am unable to download the logs for failed tests.
Most likely, they should be unrelated, but wanted to confirm. Will retry downloading and analyzing later.